### ISSUE TITLE
Feature Request: Ability to pin tooltips on the chart so that a user can make selections within the tooltip

### ISSUE THREAD

**Issue by tonydiaz** [carbon-design-system/carbon-charts#735]
#### I am submitting a...

- [x] Feature request
- [ ] Design defect
- [ ] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.34.8

### Issue description
Requesting the ability to pin a tooltip on the charts. Having the ability to pin a tooltip would allow a user to view information on the data source and take some action (click a link/button) within the tooltip without moving away from the chart. 

Within Watson assistant charts, you can select a link from the tooltip to view the specific logs for a data point. This gives the user a quick way to dive into the data for further review. In Amplitude charts, user behavior is tracked and you can select an option in the tooltip to group a set of users.  

**Amplitude:**
<img width="387" alt="Screen Shot 2020-08-04 at 7 53 46 PM" src="https://user-images.githubusercontent.com/7145340/89356686-68c92680-d68c-11ea-9dd2-77792d39f363.png">

**Watson Assistant:**
<img width="269" alt="Screen Shot 2020-08-04 at 7 53 57 PM" src="https://user-images.githubusercontent.com/7145340/89356643-50f1a280-d68c-11ea-84a5-16e73c8e06b2.png">

#### Steps to produce the issue
N/A

#### Current behavior
Currently, the ability to pin tooltips doesn't exist. There is the ability to listen to specific events on a chart but this would make the user go outside of the chart to make any selections:
https://carbon-design-system.github.io/carbon-charts/?path=/story/tutorials--event-listeners

#### Expected behavior
Pinning the tooltip should be configurable using the options property to set the flag. 

The pinning could be triggered by either the user clicking on a data point or hovering over the data point. Clearing the pinned tooltip could happen by the user clicking anywhere on the chart that isn't the tooltip.

### Screenshot or recording
**Amplitude charts:** To pin the tooltip you must click on the datapoint
![amplitude](https://user-images.githubusercontent.com/7145340/89355615-d4f65b00-d689-11ea-8423-a528dee54322.gif)

**Watson Assistant charts:** To pin the tooltip you need to hover over the tooltip
![watsonAssistant](https://user-images.githubusercontent.com/7145340/89355910-87c6b900-d68a-11ea-99ce-070b0a23541f.gif)

**Comment by jeanservaas** [comment]
@tonydiaz 

The problem with that tooltip is that it doesn’t really stylistically relate to the tooltip designs from accurat that we’re moving forward with. The design is closer to the Carbon tooltip. But then my question would be, when / why would you get a pointer and radial corners on a tooltip, and when would you not. What is the visual logic. 

<img width="634" alt="image" src="https://user-images.githubusercontent.com/43144260/89432970-5cce7a80-d707-11ea-8b9b-88598e87ace8.png">

If the tooltip has interactive components in Carbon we open it on click rather than on hover and it stays open. I see the logic for the close button in situations like this and I'd probably want to talk to @dakahn about accessibility best practices here. 

The bottom line is, Carbon doesn't have any design resources to devote to this but we wouldn't block your development of this feature enhancement for your product.

**Comment by dakahn** [comment]
That interaction would prove pretty challenging to pull off while still adhering to our requirements around keyboard and screen reader accessibility. Something to keep in mind for anyone taking on this enhancement. 

_More generally_ charts (or ["complex images" as they're referred to by the W3](https://www.w3.org/WAI/tutorials/images/complex/)) should provide a text based alternative to the displayed information (via the `longdesc` attribute). Providing text alternatives should work even for fancier charts that require some user interaction (clicking nodes/edges) to reveal information. This isn't possible though if the chart displays dynamic content so falling back to a table would be what I'd expect in those cases.

**Comment by fsedillo** [comment]
@jeanservaas Hello! I think @tonydiaz and I were more focused on the ability to add a link within the tooltip (vs the radial corner stylization). Using a flag of some sort to enable/disable the link within the tooltip could be one implementation we were thinking. We would be happy to try this enhancement ourselves to see how we could make it work with this Carbon chart and feed back into the system as a pattern.

@dakahn Could you elaborate on what you mean by displaying `dynamic content`? We are currently using data returned by an API to populate the chart.

**Comment by dakahn** [comment]
That's a great example of dynamic content. Data returned from an API couldn't be programmed into the longdesc attribute in order to make the chart accessible

**Comment by tonydiaz** [comment]
@dakahn I'm unable to see the `longdesc` on the carbon charts in the storybook. Is that something currently supported?
https://carbon-design-system.github.io/carbon-charts/?path=/story/line--line-dense-time-series

I would imagine most uses of carbon charts get their data from APIs. 

I don't think it would much effort to generate a table with the chart information and add a `longdesc`

**Comment by theiliad** [comment]
> @dakahn I'm unable to see the `longdesc` on the carbon charts in the storybook. Is that something currently supported?
> https://carbon-design-system.github.io/carbon-charts/?path=/story/line--line-dense-time-series
> 
> I would imagine most uses of carbon charts get their data from APIs.
> 
> I don't think it would much effort to generate a table with the chart information and add a `longdesc`

yes, the table representation would be coming in with the toolbar at the topright of the chart. `longdesc` we haven't used yet

**Comment by designertyler** [comment]
Without changing the styling of a tooltip or adding a close affordance like an "X" I think it seems reasonable to allow for the Watson assistant pinning behavior as an alternative. 

It sounds like there could be three different tooltip behaviors:
Default (as-is): Only get tooltip when the cursor hovers on a data point.
Click to pin: Clicking a point pins the tooltip. Click anywhere outside the tooltip to dismiss. Hover still behaves like default. 
Hover to pin (Watson assistant): Tooltip on hover is always pinned until another tooltip is hovered or the cursor hovers outside the chart.

All three of those could be possible without any visual design changes.

**Comment by dakahn** [comment]
I was speaking generally about accessible charts. Longdesc is not currently supported to my knowledge, but I haven't worked much with the charts/charting team.

It's possible to generate a longdesc from dynamic data, but probably not feasible to create something human readable and meaningful (something like "The largest bar shown is Product A. The second greatest bar is Product B but there is a a twenty percent gap between Product B and Product A..."). Keyboard accessibility, proper clear labeling, and avoiding disclosure patterns/having the user hunt for data would also all aide in usability and accessibility in chart design.
